theano/gpuarray/opt.py

```diff
@@ -210,6 +211,85 @@ def apply(self, fgraph):
                     0, 'fast_run', 'fast_compile', 'merge')
 
 
+class GraphToGPU(Optimizer):
+    """
+    Transfer the graph as a whole to GPU instead of replacing nodes
+    """
+    def add_requirements(self, fgraph):
+        fgraph.attach_feature(toolbox.ReplaceValidate())
+
+    def apply(self, fgraph):
+        mapping = {}
+        move_to_GPU = True
+
+        # Building a new graph
+        # Iterating through inputs of graph
+        for i in fgraph.inputs:
+            if isinstance(i.type, tensor.TensorType):
+                mapping[i] = GpuFromHost(None)(i)
+            else:
+                mapping[i] = i
+
+        # Iterating through output of all the nodes
+        for n in fgraph.toposort():
+            for o in n.outputs:
+                if isinstance(i.type, tensor.TensorType):
```

Comment:
To clarify the message, you have only 1 segmentation fault. But this correspond to code 11. A segmentation fault always imediatly kill the process where this happen. So you can't have more then 1 per process:)  Check this if you want to know more in detail what this is:

https://en.wikipedia.org/wiki/Segmentation_fault

But in this case, we know there is a problem when we shutdown Theano with the new back-end. As this happen at the shutdown, it is not high in the list of thing to do in it. You can just ignore it for for now. We see that tests passed.